docs/source/getting_started/benchmarks/imagenette_benchmark.py

```diff
@@ -237,6 +234,25 @@ def get_data_loaders(batch_size: int, model):
     return dataloader_train_ssl, dataloader_train_kNN, dataloader_test
 
 
+class CosineWarmupScheduler(torch.optim.lr_scheduler.LambdaLR):
+    def __init__(self, optimizer, warmup_epochs, last_epoch=-1, verbose=False):
+        self.warmup_epochs = warmup_epochs
+        super().__init__(optimizer, self.scale_lr, last_epoch, verbose=False)
+
+    def scale_lr(self, epoch):
+        if epoch < self.warmup_epochs:
+            return epoch / self.warmup_epochs
+        else:
```

Comment:
I could also do here:

`return 0.5*utils.cosine_schedule(epoch - self.warmup_epochs, max_epochs - self.warmup_epochs, 0, 1)`

But I think it hurts legibility

Reply:
It is probably better to use the existing `cosine_schedule` method as it is well tested. Also I believe you don't need the `0.5` factor.